test/cases/5.6_string_manipulation.yaml

```diff
@@ -259,6 +259,117 @@ tests:
     expression: Functions.str.empty.toChars()
     result: []
 
+  - desc: '** split string on delimiter'
+    expression: "'a,b,c'.split(',')"
+    result: ['a', 'b', 'c']
+
+  - desc: '** split string on delimiter'
+    expression: "'a,,c'.split(',')"
+    result: ['a', 'c']
+
+  - desc: '** join string with separator'
+    expression: "('a'|'b'|'c').join(',')"
+    result: ['a,b,c']
+
+  - desc: '** trim down a string'
+    expression: "'   this is a string   '.trim()"
+    result: ['this is a string']
+
+# this should run on each item in the collection with a select()
+  - desc: '** trim down a string null'
+    expression: "'a ,, c'.split(',').select(trim())"
+    result: ['a', 'c']
+
+  - desc: '** encode urlbase64'
+    expression: "'http://example.org/fhir'.encode('base64')"
+    result: ['aHR0cDovL2V4YW1wbGUub3JnL2ZoaXI=']
+
+  - desc: '** decode'
+    expression: "'aHR0cDovL2V4YW1wbGUub3JnL2ZoaXI='.decode('base64')"
+    result: ['http://example.org/fhir']
+
+# testEncodeDecode Tests from the FHIR spec
+  - desc: '** testEncodeDecode 1e'
+    expression: "'test'.encode('base64')"
+    result: ['dGVzdA==']
+
+  - desc: '** testEncodeDecode 2e'
+    expression: "'test'.encode('hex')"
+    result: ['74657374']
+
+  - desc: '** testEncodeDecode 3e'
+    expression: "'subjects?_d'.encode('base64')"
+    result: ['c3ViamVjdHM/X2Q=']
```

Comment:
It would be nice if the test string output also included + or -.

Reply:
Do you mean an additional test like this one?
``` yaml
  -dec: extra test that results with a `+` in the encoded string
    expression: "'subjects>_d'.encode('base64')"
    result: ['c3ViamVjdHM+X2Q=']
```